catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SolrClientAdaptor.java

```diff
@@ -0,0 +1,186 @@
+        this.metacardClientSupplierFunction =
+                (solrClient) -> new CacheSolrMetacardClient(solrClient,
+                        filterAdapter,
+                        solrFilterDelegateFactory);
+        this.clientAdaptorSupplierFunction = (solrClient) -> new InitializedSolrClientAdaptor(
+                solrClient);
+        this.coreName = coreName;
+        this.filterAdapter = filterAdapter;
+        this.solrClientFactory = solrClientFactory;
+        this.solrFilterDelegateFactory = solrFilterDelegateFactory;
+    }
+
+    public void init() {
+        RetryPolicy retryPolicy = new RetryPolicy();
+
+        Failsafe.with(retryPolicy)
+                .onRetry((exception) -> LOGGER.debug(
+                        "Failed to get Solr client for SolrCache. Retrying...",
+                        exception))
+                .onSuccess((result, context) -> LOGGER.debug(
+                        "SolrCache successfully initialized after {} retries",
+                        context.getExecutions()))
+                .run(this::getSolrClient);
+    }
+
+    //For unit testing purposes.
+    State getState() {
+        return state;
+    }
+
+    //For unit testing purposes.
+    void setMetacardClientSupplierFunction(
+            Function<SolrClient, CacheSolrMetacardClient> metacardClientSupplierFunction) {
+        this.metacardClientSupplierFunction = metacardClientSupplierFunction;
+    }
+
+    //For unit testing purposes.
+    void setClientAdaptorSupplierFunction(
+            Function<SolrClient, InitializedSolrClientAdaptor> clientAdaptorSupplierFunction) {
+        this.clientAdaptorSupplierFunction = clientAdaptorSupplierFunction;
+    }
+
+    /**
+     * Gets a reference to the {@link SolrMetacardClient} associated with the {@link SolrClient}.
+     */
+    SolrMetacardClient getSolrMetacardClient() {
+        return client;
+    }
+
+    /**
+     * Commits Solr transactions.
+     *
+     * @see SolrClient#commit()
+     */
+    void commit() throws SolrServerException, IOException {
+        state.commit();
+    }
+
+    /**
+     * Closes the Solr connection.
+     *
+     * @see SolrClient#close()
+     */
+    void close() throws IOException {
+        state.close();
+    }
+
+    /**
+     * Deletes Solr documents that match the query provided.
+     *
+     * @see SolrClient#deleteByQuery(String)
+     */
+    UpdateResponse deleteByQuery(String query) throws SolrServerException, IOException {
+        return state.deleteByQuery(query);
+    }
+
+    private void getSolrClient() throws InterruptedException, ExecutionException, TimeoutException {
+
+        if (solrClientFuture == null) {
+            solrClientFuture = solrClientFactory.newClient(coreName);
+        }
+
+        SolrClient solrClient = solrClientFuture.get(5, SECONDS);
+
+        if (solrClient == null) {
+            solrClientFuture = null;
+            throw new IllegalStateException();
```

Comment:
This method is being executed under a `RetryPolicy` which will retry for 10 minutes.